src/test/java/seedu/address/logic/commands/AddRecurringSessionCommandTest.java

```diff
@@ -0,0 +1,221 @@
+    }
+
+    /**
+     * A Model stub that contains a single student with a single recurring session.
+     */
+    private class ModelStubWithRecurringSession extends ModelStub {
+        private final Student student;
+
+        ModelStubWithRecurringSession(RecurringSession recurringSession) {
+            requireNonNull(recurringSession);
+            this.student = new StudentBuilder().build();
+            student.addSessions(new RecurringSessionBuilder().build()); }
+
+        @Override
+        public boolean hasStudent(Student student) {
+            requireNonNull(student);
+            return this.student.isSameStudent(student);
+        }
+
+        @Override
+        public boolean hasOverlappingSession(RecurringSession recurringSession) {
+            for (Session recurringSessionOfStudent : this.student.getListOfSessions()) {
+                if (recurringSessionOfStudent instanceof RecurringSession) {
+                    return recurringSessionOfStudent.isOverlapping(recurringSession);
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public boolean hasName(Name name) {
+            requireNonNull(name);
+            return this.student.getName().equals(name);
+        }
+
+        @Override
+        public boolean hasSession(Session session) {
+            requireNonNull(session);
+            return this.student.getListOfSessions().get(0).equals(session);
+        }
+    }
+
+    /**
+     * A Model stub that always accept the session being added.
+     */
+    private class ModelStubAcceptingRecurringSessionAdded extends ModelStub {
+        private Student oneStudent = new StudentBuilder().build();
+
+        @Override
+        public boolean hasStudent(Student student) {
+            requireNonNull(student);
+            return oneStudent.isSameStudent(student);
+        }
+
+        @Override
+        public boolean hasName(Name name) {
+            requireNonNull(name);
+            return oneStudent.getName().equals(name);
+        }
+
+        @Override
+        public boolean hasSession(Session session) {
+            requireNonNull(session);
+            List<Session> sessionList = oneStudent.getListOfSessions();
+            for (Session sessionInList : sessionList) {
+                SessionDate sessionDate = sessionInList.getSessionDate();
+                if (sessionDate.equals(session.getSessionDate())) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public boolean hasOverlappingSession(RecurringSession recurringSession) {
+            for (Session recurringSessionOfStudent : oneStudent.getListOfSessions()) {
+                if (recurringSessionOfStudent instanceof RecurringSession) {
+                    return recurringSessionOfStudent.isOverlapping(recurringSession);
+                }
+            }
+            return false;
```

Comment:
For example, the use of the actual implementation of `isOverlapping` of `RecurringSession` makes it dependent on the method.

Reply:
oh right so by right in a stub it should straight away return false?
or is this method simpy un-stub-able

Reply:
hmm then again seems like we can't do without such a dependency because we are calling the overlap check from within parser as well